packages/origin-247-certificate/src/offchain-certificate/certificate.aggregate.ts

```diff
@@ -0,0 +1,151 @@
+import {
+    ICertificateEvent,
+    CertificateEventType,
+    CertificateTransferredEvent,
+    CertificateClaimedEvent
+} from './events/Certificate.events';
+import { IIssueCommand, ICertificateReadModel } from '../types';
+import { CertificateErrors } from './errors';
+import { BigNumber } from 'ethers';
+
+export class CertificateAggregate {
+    private certificate: ICertificateReadModel;
+
+    constructor(private readonly events: ICertificateEvent[]) {
+        // events need to be sorted first
+        // field to sort by needs to be added to event
```

Comment:
I think there should be `createdAt` field on event created by repository. Also, this aggregate can make sorting by itself, just to make sure.